pkg/api/controller.go

```diff
@@ -556,6 +556,40 @@ func (c *Controller) Login(w http.ResponseWriter, r *http.Request, body apigen.L
 	writeResponse(w, r, http.StatusOK, response)
 }
 
+func (c *Controller) ExternalPrincipalLogin(w http.ResponseWriter, r *http.Request, body apigen.ExternalPrincipalLoginJSONRequestBody) {
+	ctx := r.Context()
+	if c.isExternalPrincipalNotSupported(ctx) {
+		writeError(w, r, http.StatusNotImplemented, "Not implemented")
+		return
+	}
+
+	c.LogAction(ctx, "external_principal_login", r, "", "", "")
+	principalID, err := c.Authentication.ExternalPrincipalLogin(ctx, body)
```

Comment:
let's make sure here that we explicitly override and distinguish between Unauthorized and SessionExpired. 
The clients will know to retry if expired. 
I suggest to do something like: 
```go
if errors.Is(err, kv.ErrSessionExpired) {
  err = ExternalPrincipalSessionExpired
}
```

Reply:
I'm not sure it's relevant for her, it's relevant for the implementation
I'll return here the error I've got
Unless you think I need to send a different status code (bad request instead of unauthorized?)

Reply:
we need standard API, how will lakeFS know the difference if we don't show it?

Reply:
I'm not sure how your suggestion changes lakeFS behavior.
Your suggestion look like this:
```
if errors.Is(err, ErrExpired) {
  err = ErrExpired
}
if error.Is(err, ErrUnauthorized) {
  err = ErrUnauthorized
}
```
What difference does it make?